File: Javascript/bubblesort.js
exports.sort = function(numbers) {
    let length = numbers.length;
    let swapped = false;
    for (let i = 0; i < length; i++) {
        for (let j = 0; j < length - i; j++) {
            if (numbers[j] > numbers[j + 1]) {
                let temp = numbers[j];
                numbers[j] = numbers[j + 1];
                numbers[j + 1] = temp;
                swapped = true;
            }
        }
        if (!swapped) {
            break;
        }
    }
};

/*
* Time Complexity
*  O(n^2)
* */

File: Java/src/com/ds/sort/BubbleSort.java
package com.ds.sort;

/**
 * @author Allabakash
 *
 */
public class BubbleSort implements Sorter{

	public BubbleSort() {
		System.out.println("******* BubbleSort *******");
	}
	
	/* 
	 * (non-Javadoc)
	 * @see com.ds.sort.Sorter#sort(int[])
	 */
	@Override
	public void sort(int[] array) {
		boolean swapped = false;
		for (int i = 0; i < array.length; i++) {
			for (int j = 0; j < array.length - 1; j++) {
				if (array[j] > array[j+1]) {
					int temp = array[j];
					array[j] = array[j+1];
					array[j+1] = temp;
					swapped = true;
				}
			}
			
			//Already sorted , no need to check other elements
			if (!swapped) {
				break;
			}
		}
	}
	
}


/*
* Time Complexity
*  O(n^2)
* */